### PR TITLE
Implements syscall 464

### LINKED ISSUE #490: Unimplemented syscall 464
According to one user of our discord, this is reached by God of War.
This seems to be the FreeBSD syscall thr_set_name.

### PR DESCRIPTION
**Pull request by SuchAFuriousDeath**
Will close #490.

### CODE REVIEW

**Comment by SuchAFuriousDeath** [comment]
@ultimaweapon It seems that one of your refactors broke the Mac build, because some used library features are unstable there. At least that's how I understand it.

**Comment by SuchAFuriousDeath** [comment]
Ok, that's weird, rerunning the build made it pass. Whatever.

**Comment by ultimaweapon** [comment]
> Not gonna lie, I'm a bit confused by the comment above the VProc struct. @ultimaweapon What will we do if a game calls fork? Do we expect no game to ever call fork? Is the 'threads' field of 'VProc' an equivalent of td_hash?
> 
> The reason I'm asking is, I need to know what the equivalent of tdfind is for us.

The game should never call to fork. IIRC tdfind is just a loop finding a thread with specific ID, which you can use Rust iterator to achieve the same function.

**Comment by ultimaweapon** [comment]
Seems like macOS randomly use Rust 1.73.

File: src/kernel/src/process/mod.rs [hunk 1]
```diff
@@ -409,6 +416,26 @@ impl VProc {
         Ok(SysOut::ZERO)
     }
 
+    fn sys_thr_set_name(self: &Arc<Self>, i: &SysIn) -> Result<SysOut, SysErr> {
+        let name: &str = unsafe { i.args[0].to_str(32) }?.unwrap();
```
Comment (ultimaweapon) [inline src/kernel/src/process/mod.rs:420]: The PS4 will use an empty name if this argument is null.

**Review by ultimaweapon — CHANGES_REQUESTED** [review]

File: src/kernel/src/process/mod.rs (hunk 1 above)
Comment (SuchAFuriousDeath) [reply src/kernel/src/process/mod.rs]: Fixed., thanks.

File: src/kernel/src/process/mod.rs [hunk 2]
```diff
@@ -409,6 +416,26 @@ impl VProc {
         Ok(SysOut::ZERO)
     }
 
+    fn sys_thr_set_name(self: &Arc<Self>, i: &SysIn) -> Result<SysOut, SysErr> {
+        let name: Option<&str> = unsafe { i.args[0].to_str(32) }?;
```
Comment (ultimaweapon) [inline src/kernel/src/process/mod.rs:420]: This argument is not a name.

File: src/kernel/src/process/mod.rs [hunk 3]
```diff
@@ -409,6 +416,26 @@ impl VProc {
...
+        let id: i32 = i.args[1].try_into().unwrap();
```
Comment (ultimaweapon) [inline src/kernel/src/process/mod.rs:421]: This argument is not an ID.

File: src/kernel/src/process/mod.rs [hunk 4]
```diff
@@ -74,6 +75,7 @@ impl VProc {
             dmem_container: gg.spawn(0), // TODO: Check the initial value on the PS4.
             budget: gg.spawn(None),
             limits,
+            comm: gg.spawn(None), //Find out how this is actually set
```
Comment (ultimaweapon) [inline src/kernel/src/process/mod.rs:78]: Better to prefix with `TODO:` so it is easy to search for what need to be done.

**Review by ultimaweapon — CHANGES_REQUESTED** [review]

**Comment by SuchAFuriousDeath** [comment]
Good catch. I don't know how I coud get that wrong :(.

**Review by ultimaweapon — APPROVED** [review]

**Pull request merged**